langchain/vectorstores/pinecone.py

```diff
@@ -354,16 +354,29 @@ def from_existing_index(
             pinecone.Index(index_name), embedding.embed_query, text_key, namespace
         )
 
-    def delete(self, ids: List[str], namespace: Optional[str] = None) -> None:
-        """Delete by vector IDs.
+    def delete(self,
+            ids: Optional[List[str]] = None,
+            delete_all: Optional[bool] = None,
+            namespace: Optional[str] = None,
+            filter: Optional[dict] = None,
+            **kwargs) -> Dict[str, Any]:
+        """Delete by vector IDs or filter.
         Args:
             ids: List of ids to delete.
+            filter: Dictionary of conditions to filter vectors to delete.
         """
 
-        # This is the maximum number of IDs that can be deleted
         if namespace is None:
             namespace = self._namespace
-        chunk_size = 1000
-        for i in range(0, len(ids), chunk_size):
-            chunk = ids[i : i + chunk_size]
-            self._index.delete(ids=chunk, namespace=namespace)
+
+        if delete_all:
+            return self._index.delete(delete_all=True, namespace=namespace, **kwargs)
+        elif ids is not None:
+            chunk_size = 1000
+            for i in range(0, len(ids), chunk_size):
+                chunk = ids[i : i + chunk_size]
+                return self._index.delete(ids=chunk, namespace=namespace, **kwargs)
```

Comment:
Doesn't this mean that the loop won't actually run over all chunks and break on the first one?

Reply:
that's true, great catch. Thanks!

Reply:
I actually ran CodiumAI-Agent a few times on your code (sorry, mistakenly, two times was published here, apologize),
and in one of these cases, it suggested this error 🤖

Reply:
I love the idea and execution, it's very helpful. I'd love to contribute if anything's open source!

Reply:
@coditamar this is great. how can I run on PR's regularly?

Reply:
@rlancemartin , we will try to release an option to run regularly this coming week!

Reply:
@rlancemartin , here you go:
https://github.com/Codium-ai/pr-agent
would love to discuss how to use 'LangChain' in 'pr-agent'